src/library/objects/ShowcaseExporter.js

```diff
@@ -0,0 +1,955 @@
+        this._loadImage("devmats", "_otherImages", "/assets/img/useitems/3.png", callback);
+
+    };
+
+    ShowcaseExporter.prototype.cleanUp = function () {
+        this.canvas = {};
+        this.ctx = {};
+        this.allShipGroups = {};
+        this._statsImages = {};
+        this._equipTypeImages = {};
+        this._shipImages = {};
+        this._otherImages = {};
+        this._equipCanvases = {};
+        this._equipGroupCanvases = {};
+    };
+
+    ShowcaseExporter.prototype._finish = function (dataURL, topLine) {
+        switch (parseInt(ConfigManager.ss_mode, 10)) {
+            case 0:
+                this._download(dataURL, topLine);
+                break;
+            case 1:
+                this._postToImgur(dataURL, topLine);
+                break;
+            default:
+                this._openInNewTab(dataURL, topLine);
+                break;
+        }
+    };
+
+    ShowcaseExporter.prototype._download = function (dataURL, topLine) {
+        if (enableShelfTimer) {
+            clearTimeout(enableShelfTimer);
+        }
+        var self = this;
+        var filename = ConfigManager.ss_directory + '/' + dateFormat("yyyy-mm-dd") + '_' + topLine.replace(" ", "") + ".png";
+        chrome.downloads.setShelfEnabled(false);
+        chrome.downloads.download({
+            url: dataURL,
+            filename: filename,
+            conflictAction: "uniquify"
+        }, function (downloadId) {
+            enableShelfTimer = setTimeout(function () {
+                chrome.downloads.setShelfEnabled(true);
+                enableShelfTimer = false;
+                self.cleanUp();
+                self.complete({downloadId: downloadId, filename: filename});
+            }, 100);
+        });
+    };
+
+    ShowcaseExporter.prototype._postToImgur = function (dataURL, topLine) {
+        var stampNow = Math.floor((new Date()).getTime() / 1000);
+        if (stampNow - imgurLimit > 10) {
+            imgurLimit = stampNow;
+        } else {
+            this._download(dataURL, topLine);
+            return false;
+        }
+
+        var self = this;
+        $.ajax({
+            url: 'https://api.imgur.com/3/credits',
+            method: 'GET',
+            headers: {
+                Authorization: 'Client-ID 088cfe6034340b1',
+                Accept: 'application/json'
+            },
+            success: function (response) {
+                if (response.data.UserRemaining > 10 && response.data.ClientRemaining > 100) {
+                    $.ajax({
+                        url: 'https://api.imgur.com/3/image',
+                        method: 'POST',
+                        headers: {
+                            Authorization: 'Client-ID 088cfe6034340b1',
+                            Accept: 'application/json'
+                        },
+                        data: {
+                            image: dataURL.substring(22),
```

Comment:
you can use this instead:
https://github.com/KC3Kai/KC3Kai/pull/1803/files#diff-4834798524872cdd6bda1e77051ff285R185